src/renderer/components/CanvasVisualiser/Events.js

```diff
@@ -1,91 +1,112 @@
+          highlight: edge.color.highlight,
+          hover: edge.color.hover,
+        },
+      }));
   this.updateEdge(edgesToUpdate);
   this.updateNode(nodesToUpdate);
 }
 
 export function setDefaultColoursOnDimmedElements(connectedNodes, connectedEdges) {
   const nodesToUpdate =
-  this.getNode()
-    .filter(x => !connectedNodes.includes(x.id))
-    .map(x => ({ id: x.id, color: x.colorClone, font: x.fontClone }))
-    .reduce(collect, []);
+    this.getNode()
+      .filter(x => !connectedNodes.includes(x.id))
+      .map(x => ({ id: x.id, color: x.colorClone, font: x.fontClone }));
 
   const edgesToUpdate =
-  this.getEdge()
-    .filter(x => !connectedEdges.includes(x.id))
-    .map(edge => ({ id: edge.id,
-      color: {
-        color: edge.color.color.replace('0.2)', '1)'),
-        highlight: edge.color.highlight,
-        hover: edge.color.hover,
-      } })).reduce(collect, []);
+    this.getEdges()
+      .filter(x => !connectedEdges.includes(x.id))
+      .map(edge => ({
+        id: edge.id,
+        color: {
+          color: edge.color.color.replace('0.2)', '1)'),
+          highlight: edge.color.highlight,
+          hover: edge.color.hover,
+        },
+      }));
 
   this.updateNode(nodesToUpdate);
   this.updateEdge(edgesToUpdate);
 }
 
 export function showEdgeLabels(edges) {
-  const edgesToUpdate =
-  edges.map(id => this.getEdge(id))
-    .map(edge => ({ id: edge.id, arrows: { to: { enabled: true, scaleFactor: 0.3 } }, label: edge.hiddenLabel }))
-    .reduce(collect, []);
+  const edgesToUpdate = edges.map(edge => ({ ...edge, label: edge.hiddenLabel }));
+  this.updateEdge(edgesToUpdate);
+}
 
+export function showEdgeArrows(edges) {
+  const edgesToUpdate = edges.map(edge => ({ ...edge, arrows: { to: { enabled: true, scaleFactor: 0.3 } } }));
   this.updateEdge(edgesToUpdate);
 }
 
 export function hideEdgeLabels(edges) {
-  const edgesToUpdate =
-  edges.map(id => this.getEdge(id))
-    .map(edge => ({ id: edge.id, arrows: { to: { enabled: false } }, label: '' }))
-    .reduce(collect, []);
+  const edgesToUpdate = edges.map(edge => ({ ...edge, label: '' }));
+  this.updateEdge(edgesToUpdate);
+}
 
+export function hideEdgeArrows(edges) {
+  const edgesToUpdate = edges.map(edge => ({ ...edge, arrows: { to: { enabled: false } } }));
   this.updateEdge(edgesToUpdate);
 }
 
 export function removeLabelsOnNotConncetedEdges(nodeId) {
   const connectedEdges = this.getNetwork().getConnectedEdges(nodeId);
-  const edgesToUpdate = this.getEdge()
+  const edgesToUpdate = this.getEdges()
     .filter(x => !connectedEdges.includes(x.id))
-    .map(edge => ({ id: edge.id, arrows: { to: { enabled: false } }, label: '' }))
-    .reduce(collect, []);
+    .map(edge => ({ ...edge, label: '' }));
 
   this.updateEdge(edgesToUpdate);
 }
 
-export function showLabelsOnEdgesConnectedToNode(nodeId) {
+export function removeArrowsOnNotConncetedEdges(nodeId) {
   const connectedEdges = this.getNetwork().getConnectedEdges(nodeId);
```

Comment:
I see you're correctly changing `connectedEdges` to `connectedEdgesId`, please make this change globally so we don't forget